contract/r/gnoswap/pool/store.gno

```diff
@@ -0,0 +1,144 @@
+package pool
+
+import (
+	"gno.land/p/gnoswap/store"
+	"gno.land/p/nt/avl"
+	"gno.land/p/nt/ufmt"
+)
+
+// StoreKey defines the keys used for storing pool data in the KV store.
+// These keys are prefixed with the domain address to ensure namespace isolation.
+type StoreKey string
+
+const (
+	// Pool data storage keys
+	StoreKeyPools                StoreKey = "pools"                // AVL tree containing all pools
+	StoreKeyFeeAmountTickSpacing StoreKey = "feeAmountTickSpacing" // Fee tier to tick spacing mapping
+	StoreKeySlot0FeeProtocol     StoreKey = "slot0FeeProtocol"     // Protocol fee percentage
+
+	// Protocol fee storage keys
+	StoreKeyPoolCreationFee StoreKey = "poolCreationFee" // Pool creation fee amount
+	StoreKeyWithdrawalFee   StoreKey = "withdrawalFee"   // Withdrawal fee in basis points
+)
+
+// poolStore implements the IPoolStore interface for pool domain storage.
+// It provides type-safe access to pool data stored in the underlying KV store.
+type poolStore struct {
+	kvStore store.KVStore
+}
+
+// GetPools retrieves the AVL tree containing all pool data.
+// This is the main data structure that stores all pool instances.
+func (s *poolStore) GetPools() *avl.Tree {
+	result, err := s.kvStore.Get(s.makeKey(StoreKeyPools))
+	if err != nil {
+		panic(err)
+	}
```

Comment:
this throwing a panic immediately when accessing unintialized storage. Although a default pool is create upon deployment, it might be worth considering adding a safeguard that creates a new one if it's not found.